### PR TITLE
openjpeg: 2.4.0 -> 2.5.0

### PR DESCRIPTION
**Pull request by SuperSandro2000**
###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/libraries/openjpeg/default.nix [hunk 1]
```diff
@@ -23,7 +21,7 @@ in
 
 stdenv.mkDerivation rec {
   pname = "openjpeg";
-  version = "2.4.0"; # don't forget to change passthru.incDir
+  version = "2.5.0";
 
   src = fetchFromGitHub {
```
Comment (ck3d) [inline pkgs/development/libraries/openjpeg/default.nix:26]: please update sha256

Comment (SuperSandro2000) [reply pkgs/development/libraries/openjpeg/default.nix:18]: oh, yeah

File: pkgs/development/libraries/openjpeg/default.nix [hunk 2]
```diff
@@ -1,90 +1,63 @@
...
-    (mkFlag mj2Support "BUILD_MJ2")
-    (mkFlag jpwlLibSupport "BUILD_JPWL")
+    "-DBUILD_THIRDPARTY=OFF"
     (mkFlag jpipLibSupport "BUILD_JPIP")
     (mkFlag jpipServerSupport "BUILD_JPIP_SERVER")
-    #(mkFlag opjViewerSupport "BUILD_VIEWER")
-    "-DBUILD_VIEWER=OFF"
+    (mkFlag opjViewerSupport "BUILD_VIEWER")
     (mkFlag openjpegJarSupport "BUILD_JAVA")
-    (mkFlag jp3dSupport "BUILD_JP3D")
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/openjpeg/default.nix:56]: mj2, jpwl and jp3d are deprecated https://github.com/uclouvain/openjpeg#deprecated-components

**Comment by SuperSandro2000** [comment]
@ofborg eval

**Comment by trofi** [comment]
commit 34dbaede64a094d59b04a8e76f366f4da4f396c `openjpeg: 2.4.0 -> 2.5.0` broke `poppler` in `staging`:

```
$ nix build -f. -L poppler
...
poppler-glib> -- Configuring done
poppler-glib> CMake Error in CMakeLists.txt:
poppler-glib>   Imported target "openjp2" includes non-existent path
poppler-glib>     "/nix/store/r74nd1znbb1vd061zj8d6g9yz56hnjw9-openjpeg-2.5.0/include/openjpeg-2.5"
poppler-glib>   in its INTERFACE_INCLUDE_DIRECTORIES.  Possible reasons include:
poppler-glib>   * The path was deleted, renamed, or moved to another location.
poppler-glib>   * An install or uninstall procedure did not complete successfully.
poppler-glib>   * The installation package was faulty and references files it does not
poppler-glib>   provide.
```

**Comment by SuperSandro2000** [comment]
The CMake files of openjpeg2 are wrong.

**Comment by SuperSandro2000** [comment]
I cherry-picked some patches from upstream and now it works again. Creating a PR.

**Comment by SuperSandro2000** [comment]
#192644

**Pull request merged**